api/src/main/java/com/epam/pipeline/manager/git/github/GitHubService.java

```diff
@@ -0,0 +1,412 @@
+                                      final String message) {
+        if (ListUtils.emptyIfNull(sourceItemVOList.getItems()).stream()
+                .anyMatch(sourceItemVO -> StringUtils.isNotBlank(sourceItemVO.getPreviousPath()))) {
+            throw new UnsupportedOperationException(NOT_SUPPORTED_PATTERN);
+        }
+        final GitHubClient client = getClient(pipeline);
+
+        ListUtils.emptyIfNull(sourceItemVOList.getItems())
+                .forEach(sourceItemVO -> client.updateFile(sourceItemVO.getPath(), sourceItemVO.getContents(),
+                message, pipeline.getBranch()));
+        return new GitCommitEntry();
+    }
+
+    @Override
+    public GitCommitEntry uploadFiles(final Pipeline pipeline, final List<UploadFileMetadata> files,
+                                      final String message) {
+        Assert.isTrue(files.size() == 1, String.format(NOT_SUPPORTED_PATTERN, "Multiple files upload"));
+        final UploadFileMetadata file = files.get(0);
+        final GitHubClient client = getClient(pipeline);
+        client.createFile(file.getFileName(), file.getBytes(), message, pipeline.getBranch());
+        return new GitCommitEntry();
+    }
+
+    @Override
+    public boolean fileExists(final Pipeline pipeline, final String filePath) {
+        return fileExists(getClient(pipeline), filePath);
+    }
+
+    private static String getContentSha(final String rawPath, final String version, final GitHubClient client) {
+        final File folder = new File(rawPath);
+        final String parentDirectory = Optional.ofNullable(folder.getParent()).orElse(Strings.EMPTY);
+        final List<GitHubContent> contents = client.getContents(parentDirectory, version);
+        return contents.stream()
+                .filter(c -> c.getPath().equals(rawPath))
+                .findFirst()
+                .map(GitHubContent::getSha)
+                .orElseThrow(() -> new RuntimeException(String.format("Source not found %s", parentDirectory)));
+    }
+
+    private GitHubClient getClient(final Pipeline pipeline) {
+        return getClient(pipeline.getRepository(), pipeline.getRepositoryToken());
+    }
+
+    private GitHubClient getClient(final String repositoryPath, final String token) {
+        return buildClient(repositoryPath, token);
+    }
+
+    private GitHubClient buildClient(final String repositoryPath, final String token) {
+        final GitRepositoryUrl repositoryUrl = GitRepositoryUrl.from(repositoryPath);
+        final String projectName = repositoryUrl.getNamespace().orElseThrow(() -> buildUrlParseError(PROJECT_NAME));
+        final String repositoryName = repositoryUrl.getProject().orElseThrow(() -> buildUrlParseError(REPOSITORY_NAME));
+        final String protocol = repositoryUrl.getProtocol();
+        final String host = repositoryUrl.getHost();
+        final String bitbucketHost = protocol + "api." + host;
```

Comment:
`bitbucketHost` is unexpected var name for this class. Let's use appropriate naming. 
Maybe we shall consider extracting common method for building git clients?

Reply:
The methos is not absolutely common, there are some differences for clients, that's why I haven't extracted it